src/H5FDros3.c

```diff
@@ -599,6 +608,240 @@ H5FD__ros3_fapl_free(void *_fa)
+{
+    char **value     = (char **)&_value;
+    herr_t ret_value = SUCCEED;
+
+    FUNC_ENTER_PACKAGE
+
+#if ROS3_DEBUG
+    HDfprintf(stdout, "H5FD__ros3_str_token_copy() called.\n");
+#endif
+
+    if (*value)
+        if (NULL == (*value = HDstrdup(*value)))
+            HGOTO_ERROR(H5E_RESOURCE, H5E_CANTALLOC, FAIL, "can't copy string property token")
+
+done:
+    FUNC_LEAVE_NOAPI(ret_value);
+} /* H5FD__ros3_str_token_copy() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5FD__ros3_str_token_cmp()
+ *
+ * Purpose:     Compares two token strings with each other.
+ *
+ * Return:
+ *     - Equivalent:     0
+ *     - Not Equivalent: non-zero value
+ *
+ * Programmer:  Jan-Willem Blokland
+ *              2023-05-26
+ *
+ *-------------------------------------------------------------------------
+ */
+static int
+H5FD__ros3_str_token_cmp(const void *_value1, const void *_value2, size_t size)
+{
+    char const *value1    = (char const *)_value1;
+    char const *value2    = (char const *)_value2;
+    int         ret_value = SUCCEED;
+
+    FUNC_ENTER_PACKAGE_NOERR
+
+    if (value1) {
+        if (value2)
+            ret_value = HDstrcmp(value1, value2);
+        else
+            ret_value = 1;
+    }
+    else {
+        if (value2)
+            ret_value = -1;
+        else
+            ret_value = 0;
+    }
+
+    FUNC_LEAVE_NOAPI(ret_value)
+} /* H5FD__ros3_str_token_cmp */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5FD__ros3_str_token_close()
+ *
+ * Purpose:     Closes/frees the memory associated to the token string.
+ *              Currently, it is an empty implementation since there no
+ *              additional treatment needed for this property.
+ *
+ * Return:      SUCCEED/FAIL
+ *
+ * Programmer:  Jan-Willem Blokland
+ *              2023-05-26
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5FD__ros3_str_token_close(const char *name, size_t size, void *value)
+{
+    herr_t ret_value = SUCCEED;
+
+    FUNC_ENTER_PACKAGE_NOERR
+
+    FUNC_LEAVE_NOAPI(ret_value)
+} /* H5FD__ros3_str_token_close */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5FD__ros3_str_token_delete()
+ *
+ * Purpose:     Deletes the property token from the property list and frees
+ *              the memory associated to the token string.
+ *              Currently, it is an empty implementation since there no
+ *              additional treatment needed for this property.
+ *
+ * Return:      SUCCEED/FAIL
+ *
+ * Programmer:  Jan-Willem Blokland
+ *              2023-05-26
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5FD__ros3_str_token_delete(hid_t prop_id, const char *name, size_t size, void *value)
+{
+    herr_t ret_value = SUCCEED;
+
+    FUNC_ENTER_PACKAGE_NOERR
```

Comment:
Similar to the way in which we do this for string properties elsewhere, I'd recommend that `H5Pset_fapl_ros3_token` should make a copy of the token string with strdup and set that pointer on the property list. That allows the caller to free the string without stale pointers being left around in any FAPL the property is set on. That would mean that this callback and `H5FD__ros3_str_token_close` would need implementations where they just free the memory for the string after retrieving the pointer for it.